app/controllers/projects_controller.rb

```diff
@@ -82,6 +82,18 @@ def deploy_group_versions
     render json: deploy_group_versions
   end
 
+  def find_via_repository_url
+    repository_url = params.require(:url)
+    projects = Project.where(
+      Project.arel_table[:repository_url].matches("%#{ActiveRecord::Base.send(:sanitize_sql_like, repository_url)}%")
+    )
+    if projects.present?
+      render json: projects.as_json, status: :ok
```

Comment:
since this works exactly like `index` support addition options there instead of building something new ?